src/fam.rs

```diff
@@ -409,15 +422,50 @@ impl<T: Default + FamStruct> FamStructWrapper<T> {
     }
 }
 
-impl<T: Default + FamStruct> PartialEq for FamStructWrapper<T> {
+impl<T: Default + FamStruct + PartialEq> PartialEq for FamStructWrapper<T> {
     fn eq(&self, other: &FamStructWrapper<T>) -> bool {
-        self.as_slice() == other.as_slice()
+        // This is safe as long as the requirements for the `FamStruct` trait to be safe are met
+        // (the implementing type and the entries elements are POD, therefore `Copy`, so memory
+        // safety can't be violated by the ownership of `fam_struct`s). It is also safe because
+        // we're trying to read `T`s from `&T`s that are pointing to properly initialized and
+        // aligned `T`s.
+        let (fam_struct, other_fam_struct): (T, T) = unsafe {
+            (
+                std::ptr::read(self.as_fam_struct_ref()),
+                std::ptr::read(other.as_fam_struct_ref()),
+            )
+        };
+        fam_struct == other_fam_struct && self.as_slice() == other.as_slice()
     }
 }
 
 impl<T: Default + FamStruct> Clone for FamStructWrapper<T> {
     fn clone(&self) -> Self {
-        FamStructWrapper::from_entries(self.as_slice())
+        // The number of entries (self.as_slice().len()) can't be > T::max_len() since `self` is a
+        // valid `FamStructWrapper`.
+        let required_mem_allocator_capacity =
+            FamStructWrapper::<T>::mem_allocator_len(self.as_slice().len());
```

Comment:
Should we clone the source FamStruct exactly by using "self.capacipy" instead of "self.as_slice().len()"?

Reply:
After having a chat with @alexandruag it turned out it would be better to avoid this kind of confusion and he suggested to add the `Clone` trait bound to `FamStruct`; this way we would simplify things a lot by simply cloning the `mem_allocator`. Does anyone have anything against this or see a potential problem with it?